src/amo/components/LandingPage/index.js

```diff
@@ -63,43 +69,49 @@ export class LandingPageBase extends React.Component {
     this.setViewContextType();
   }
 
-  componentDidUpdate() {
-    const { params } = this.props;
+  componentWillReceiveProps(nextProps) {
+    const { params } = nextProps;
+
     if (!apiAddonTypeIsValid(params.visibleAddonType)) {
-      log.warn(oneLine`Skipping componentDidUpdate() because visibleAddonType
-        is invalid: ${params.visibleAddonType}`);
+      log.warn(oneLine`Skipping componentWillReceiveProps() because
+        visibleAddonType is invalid: ${params.visibleAddonType}`);
       return;
     }
 
-    this.getLandingDataIfNeeded();
-    this.setViewContextType();
+    this.getLandingDataIfNeeded(nextProps);
+    this.setViewContextType(nextProps);
   }
 
-  getLandingDataIfNeeded() {
+  getLandingDataIfNeeded(nextProps = {}) {
     const {
       addonTypeOfResults,
-      dispatch,
       errorHandler,
       loading,
       params,
       resultsLoaded,
-    } = this.props;
+    } = {
+      ...this.props,
+      ...nextProps,
```

Comment:
If I comment this out all tests pass so the updating logic needs better coverage.

Reply:
I have added a test case to cover this case.